integration_tests/pubsub_test.go

```diff
@@ -0,0 +1,415 @@
+			Signer:            orch1.address().String(),
+		}
+
+		_, err = s.chain.sendMsgs(*orch1ClientCtx, &removeSubscriberIntent1Msg)
+		s.Require().NoError(err)
+		s.T().Log("RemoveSubscriberIntent for orch 1 submitted correctly")
+
+		s.T().Log("Verifying SubscriberIntent for orch 1 removed")
+		subscriberIntentsResponse, err = pubsubQueryClient.QuerySubscriberIntents(context.Background(), &types.QuerySubscriberIntentsRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(subscriberIntentsResponse.SubscriberIntents, 0)
+
+		// delete subscribers
+		s.T().Log("Removing Subscriber for orch 0")
+		removeSubscriber0Msg := types.MsgRemoveSubscriberRequest{
+			SubscriberAddress: orch0.address().String(),
+			Signer:            orch0.address().String(),
+		}
+
+		_, err = s.chain.sendMsgs(*orch0ClientCtx, &removeSubscriber0Msg)
+		s.Require().NoError(err)
+		s.T().Log("RemoveSubscriber for orch 0 submitted correctly")
+
+		s.T().Log("Verifying Subscriber for orch 0 removed")
+		subscribersResponse, err = pubsubQueryClient.QuerySubscribers(context.Background(), &types.QuerySubscribersRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(subscribersResponse.Subscribers, 1)
+		s.Require().Equal(subscribersResponse.Subscribers[0].Address, orch1.address().String())
+
+		s.T().Log("Removing Subscriber for orch 1")
+		removeSubscriber1Msg := types.MsgRemoveSubscriberRequest{
+			SubscriberAddress: orch1.address().String(),
+			Signer:            orch1.address().String(),
+		}
+
+		_, err = s.chain.sendMsgs(*orch1ClientCtx, &removeSubscriber1Msg)
+		s.Require().NoError(err)
+		s.T().Log("RemoveSubscriber for orch 1 submitted correctly")
+
+		s.T().Log("Verifying Subscriber for orch 1 removed")
+		subscribersResponse, err = pubsubQueryClient.QuerySubscribers(context.Background(), &types.QuerySubscribersRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(subscribersResponse.Subscribers, 0)
+
+		// remove default subscription prop
+		s.T().Log("Creating RemoveDefaultSubscriptionProposal")
+		removeDefaultSubscriptionProp := types.RemoveDefaultSubscriptionProposal{
+			Title:          "remove a default subscription",
+			Description:    "a default subscription is being removed!",
+			SubscriptionId: subscriptionID,
+		}
+
+		removeDefaultSubscriptionPropMsg, err := govtypesv1beta1.NewMsgSubmitProposal(
+			&removeDefaultSubscriptionProp,
+			sdk.Coins{
+				{
+					Denom:  testDenom,
+					Amount: math.NewInt(2),
+				},
+			},
+			proposer.address(),
+		)
+		s.Require().NoError(err, "Unable to create governance proposal")
+
+		s.submitAndVoteForProposal(proposerCtx, orch0ClientCtx, propID, removeDefaultSubscriptionPropMsg)
+		propID++
+
+		s.T().Log("Verifying DefaultSubscription correctly removed")
+		defaultSubscriptionsResponse, err = pubsubQueryClient.QueryDefaultSubscriptions(context.Background(), &types.QueryDefaultSubscriptionsRequest{})
+		s.Require().NoError(err)
+		s.Require().Len(defaultSubscriptionsResponse.DefaultSubscriptions, 0)
+
```

Comment:
I think we're missing removal of the publisher intent? Which raises the question, should the RemovePublisher handler delete the corresponding publisher intents? Same question for subscribers.

Reply:
Ah yes you're right, I forgot to remove the PublisherIntent.

I believe the code is supposed to delete Intents when either a Publisher or Subscriber is deleted, and also remove SubscriberIntents when a PublisherIntent is deleted, but I hadn't included that in this test yet. It's a good idea to do.